package/MDAnalysis/core/groups.py

```diff
@@ -2554,26 +2564,69 @@ def write(self, filename=None, file_format="PDB",
             ``"conect"``: write only the CONECT records defined in the original
             file. ``"all"``: write out all bonds, both the original defined and
             those guessed by MDAnalysis. ``None``: do not write out bonds.
-            Default os ``"conect"``.
+            Default is ``"conect"``.
+        frames:
+            An ensemble of frames to write. The ensemble can be an list or
+            array of frame indices, a mask of booleans, an instance of
+            :class:`slice`, or the value returned when a trajectory is indexed.
+            By default, 'frames' is set to ``None`` and only the current frame
+            is written.
 
 
         .. versionchanged:: 0.9.0 Merged with write_selection. This method can
             now write both selections out.
+        .. versionchanged:: 0.19.0
+            Can write multiframe trajectories with the 'frames' argument.
         """
         # check that AtomGroup actually has any atoms (Issue #434)
         if len(self.atoms) == 0:
             raise IndexError("Cannot write an AtomGroup with 0 atoms")
 
         trj = self.universe.trajectory  # unified trajectory API
-
-        if trj.n_frames == 1:
-            kwargs.setdefault("multiframe", False)
+        if frames is None or frames == 'all':
```

Comment:
We should remember the current frame and once everything is done, reset to the current frame. Otherwise the trajectory changes depending on what you write.